static/app/views/detectors/utils/useDetectorFilterKeys.tsx

```diff
@@ -0,0 +1,40 @@
+import {useMemo} from 'react';
+
+import type {TagCollection} from 'sentry/types/group';
+import useAssignedSearchValues from 'sentry/utils/membersAndTeams/useAssignedSearchValues';
+import {DETECTOR_FILTER_KEYS} from 'sentry/views/detectors/constants';
+
+export function useDetectorFilterKeys(): TagCollection {
+  const assignedValues = useAssignedSearchValues();
+
+  return useMemo(() => {
+    return Object.fromEntries(
+      Object.keys(DETECTOR_FILTER_KEYS).map(key => {
+        const {values} = DETECTOR_FILTER_KEYS[key] ?? {};
+
+        // Special handling for assignee field to provide user/team values
+        if (key === 'assignee') {
+          return [
+            key,
+            {
+              key,
+              name: key,
+              predefined: true,
+              values: assignedValues,
+            },
+          ];
+        }
+
+        return [
+          key,
+          {
+            key,
+            name: key,
+            predefined: values !== undefined,
+            values,
+          },
+        ];
+      })
+    );
+  }, [assignedValues]);
+}
```

Comment:
nit: felt like it was a little more concise using `entries` rather than keys and then inlining the checks for `isAssignee`.
```js
export function useDetectorFilterKeys(): TagCollection {
  const assignedValues = useAssignedSearchValues();

  return useMemo(() => {
    return Object.fromEntries(
      Object.entries(DETECTOR_FILTER_KEYS).map(([key, config]) => {
        const { values } = config ?? {};
        const isAssignee = key === 'assignee';
        
        return [key, {
          key,
          name: key,
          predefined: isAssignee || values !== undefined,
          values: isAssignee ? assignedValues : values,
        }];
      })
    );
  }, [assignedValues]);
}
```